src/util/type-utils.ts

```diff
@@ -191,3 +191,7 @@ export type NarrowPartial<S, T> = {
 }
 
 export type SqlBool = boolean | 0 | 1
+
+export interface Type<T> extends Function {
```

Comment:
This is well-know utility type used by many typescript projects such as angular/nestjs. Since I don't want to be stubborn, I will change it to Constructor if you want, but please reconsider.

[Sourcegraph](https://sourcegraph.com/search?q=context:global+interface+Type%3CT%3E+lang:typescript+&patternType=standard&sm=1&groupBy=repo)

Reply:
https://sourcegraph.com/search?q=context:global+interface+Constructor%3CT%3E+lang:typescript+&patternType=standard&sm=1&groupBy=repo

Reply:
Haha the winner is Typescript